launcher/ui/MainWindow.cpp

```diff
@@ -1351,10 +1352,11 @@ void MainWindow::on_actionDeleteInstance_triggered()
 
     if (APPLICATION->instances()->trashInstance(id)) {
         ui->actionUndoTrashInstance->setEnabled(APPLICATION->instances()->trashedSomething());
-        return;
+    } else {
+        APPLICATION->instances()->deleteInstance(id);
     }
-
-    APPLICATION->instances()->deleteInstance(id);
+    APPLICATION->settings()->set("SelectedInstance", QString());
+    selectionBad();
```

Comment:
This will change the behaviour - previously the instance before the deleted instance would be selected.

Reply:
No, it doesn't change the current behavior. 
In that setting, we already have an invalid id(the id of the deleted instance).
So it doesn't matter if I rewrite it with an empty string.
Also, selectionBad will put all the other stuff to more generic values and then tries again to select a new instance.
I did try side by side to delete an instance on both the current develop and this version and once deleted it selects a new instance.

Reply:
:o I see